app/models/contract.rb

```diff
@@ -86,9 +87,11 @@ def self.issued_by(issuer, *ids)
     where.has { name.op('COLLATE', sql(collate)).matches(pattern)}
   }
 
-  scope :by_account, lambda { |account| where({ :user_account_id => account.id } ) }
+  scope :by_account, ->(account) { where.has { user_account_id == account } }
   scope :by_account_query, lambda { |query| where( { :user_account_id => Account.buyers.search_ids(query) } ) }
 
+  scope :not_paid_since, ->(unpaid_time = MAX_UNPAID_TIME.ago) { where.has { (paid_until >= unpaid_time) | (variable_cost_paid_until >= unpaid_time) } }
```

Comment:
I cannot really understand this, if I read correctly, not paid since the 1st december is:

- fixed cost was paid at least until 1st december
- variable cost was paid at least until 1st december

Reply:
I don't understand where you see the 1st December 😕 6 months ago is last July, but leaving that aside, let's go to the logic:

If I understood correctly from what @guicassolato told me (he should review this as well):
  - `paid_until` is the last time that this contract paid for fixed cost.
  - `variable_cost_paid_until`  is the last time that this contract paid for variable cost.
Which means that unless one of them has a recent date (before 6 months), it is unpaid.
That also means that this is wrong so I will push the fix and a better test 👍 Good catch 👍

Reply:
Well you can pass any date so I just used 1st december